packages/backend-data/API.md

```diff
@@ -15,14 +15,25 @@ export type ApiKeyAuthorizationModeProps = {
     expiresInDays?: number;
 };
 
+// @public
+export const AuthorizationModeMapping: {
+    readonly iam: "AWS_IAM";
+    readonly userPool: "AMAZON_COGNITO_USER_POOLS";
+    readonly oidc: "OPENID_CONNECT";
+    readonly apiKey: "API_KEY";
+    readonly lambda: "AWS_LAMBDA";
+};

```

Comment:
can you define this as `enum AuthorizationMode` ? or `DataAuthorizationMode` ?

Btw. regarding "Continues to honor the CDK modes while adding a deprecation warning when they are used" from PR description.
We're in preview mode, we should just drop them and map this internally in implementation and make types clean.

Reply:
I've reached out to Rene re: "We're in preview mode, we should just drop them"

I'm not quite following the thought to express this mapping as an enum. I think for customers to use string literals, the enum would have to be mapped to the new names, but then I'm unclear. Can you pose an example of what this would look like?

Reply:
That type which was here before looks like this enum https://github.com/aws-amplify/amplify-backend/blob/c28d536f5def56aff6b0f7effe2cf9dc7485dae8/packages/client-config/src/client-config-types/client_config.ts#L16-L22 .